### PR TITLE
replicate submodels that aren't associated with subsystems

### LINKED ISSUE #3967: Bug: Submodel rotation LOD updating only occurs on subsystems
The submodel update function needs to be applied to all submodels, not just subsystems.
This way, the new animation system can properly work across LODs

### PR DESCRIPTION
**Pull request by Goober5000**
There are a few related improvements here:

1) Add a new function for iterating over the submodel hierarchy
2) Add a counterpart to the `subobj_num` field so that a submodel can refer to its ship_info subsystem
3) Update `ship_model_replicate_submodels` to replicate all movable submodels, even the ones that aren't associated with subsystems

Fixes #3967

### CODE REVIEW

File: code/ship/ship.cpp [hunk 1]
```diff
@@ -13552,6 +13559,24 @@ void ship_model_replicate_submodels(object *objp)
...
+			auto sm = &pm->submodel[submodel];
+
+			// skip submodels that belong to subsystems since we already updated them above
+			// (Turrets are tricky because they have two submodels per subsystem, but because the subsystem index is copied
+			// when the model is read, both submodels are correctly skipped here.)
+			if (sm->subsys_num >= 0)
+				return;
+
+			// the only non-subsystem submodels that need to be updated are the ones that can move
+			if (!sm->flags[Model::Submodel_flags::Can_move])
```
Comment (BMagnu) [inline code/ship/ship.cpp:13575]: This assumption is not true anymore.
The $triggered property is no longer required due to the way that animations step over time.
Hence, an animated submodel might not have this flag set.

Comment (Goober5000) [reply code/ship/ship.cpp:13575]: That flag needs to be set though.  Rotating subsystems set that flag, and so will translating subsystems once that's finished.  Plenty of places in the code use that flag to determine whether they should treat submodels as static or moveable: dockpoints, thrusters, etc.

In fact I just realized the collision code needs to be fixed to use it too; it currently uses an alternate method that won't capture all situations.

Comment (Goober5000) [reply code/ship/ship.cpp:13573]: Added the flag to the collision code.

Comment (BMagnu) [reply code/ship/ship.cpp:13573]: #3972 addresses the issue that not all animated subsystems can have Can_move

**Review by BMagnu — APPROVED** [review]

**Pull request merged**